LiveChat/components/Chat.js

```diff
@@ -134,20 +40,25 @@ export default class Chat extends React.Component {
     if (this.props.isChatOn) {
       return (
         <View
-          animation="lightSpeedIn"
+          animation="fadeInUp"
           style={styles.container}
           ref={(ref) => { this.chat = ref; }}
         >
           <NavigationBar chatTitle={this.props.chatTitle} closeChat={this.closeChat} />
-          <Text style={styles.status}>
-            { this.state.onlineStatus ? this.props.greeting : this.props.noAgents }
-          </Text>
+          { this.props.connectionState !== 'connected' && <Text style={styles.connectionStatus}>Reconnecting...</Text> }
+          {this.props.headerText && <Text style={styles.status}>{ this.props.headerText }</Text> }
           <GiftedChat
-            messages={this.state.messages}
+            inverted={false}
+            messages={this.props.messages}
+            scrollToBottom
             renderFooter={this.renderFooter}
             onSend={this.handleSend}
-            onInputTextChanged={this.handleInputTextChange}
-            user={this.getVisitor()}
+            onInputTextChanged={this.props.onInputChange}
+            user={this.props.customer}
+            isTyping={this.props.isTyping}
+            onQuickReply={this.props.onQuickReply}
+            disableComposer={this.props.disableComposer}
+            showAvatarForEveryMessage={false}
             {...this.props}
```

Comment:
By spreading whole `this.props` the object here you are effectively "double passing" props passed earlier.
In order to avoid this behaviour, it will be best to deconstruct needed values from `this.props` object and use `rest operator` to grab all not used values and then spread it here.